### PR TITLE
Added a mechanism for caching and error handling

### PR DESCRIPTION
**Pull request by Shaked-g**
caches the url and the vscode.Hover so that Axios is only called for new urls.
Works in debug mode, you need to check that it works in production for the vscode workspace and clears the cache when vscode closes.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ export function activate(context: vscode.ExtensionContext) {
   context.subscriptions.push(vscode.languages.registerHoverProvider({ scheme: "file" }, { provideHover }));
 }
 
+// A Map to store cached hover content
+const hoverContentCache: Map<string, vscode.Hover> = new Map();
```
Comment (baruchiro) [inline src/extension.ts:18]: You can remove the comments


```suggestion

const hoverContentCache: Map<string, vscode.Hover> = new Map();
```

File: src/extension.ts [hunk 2]
```diff
@@ -23,11 +26,29 @@ const provideHover = async (document: vscode.TextDocument, position: vscode.Posi
...
+  }
+
+  try {
+    const response = await axios.get(url);
+    const meta = getMetaTags(response.data);
+    const hoverContent = formatHover(meta);
+
+    // Cache hover content for the URL
+    const vsHoverContent = new vscode.Hover(hoverContent);
+    hoverContentCache.set(url, vsHoverContent);
```
Comment (baruchiro) [inline src/extension.ts:43]: Consider extracting all the cache mechanism to another file, with function like `getOrCreate(key, () => { /* generate new value */ })`

File: src/extension.ts [hunk 3]
```diff
@@ -57,5 +78,8 @@ const calculateLengthOfImage = (rows: string[]) => {
   return Math.min(longestRow.length * 7, maxImageWidth);
 };
 
+
+
```
Comment (baruchiro) [inline src/extension.ts:82]: Pull the latest `master` changes and run `yarn && yarn prettier --write src/extension.ts`

**Review by baruchiro — COMMENTED** [review]
Thank you!!

**Comment by baruchiro** [comment]
Oh I didn't finished the review

**Comment by baruchiro** [comment]
@Shaked-g tell me if you plan to continue this or I will take it

**Comment by Shaked-g** [comment]
@baruchiro Thank you for the reminder, ill continue this.

File: src/extension.ts (hunk 2 above)
Comment (Shaked-g) [reply src/extension.ts]: I separated to a get and set functions.
a getOrCreate will require to move the whole axios get process there too.

**Comment by Shaked-g** [comment]
@baruchiro, i committed the changes

**Review by baruchiro — APPROVED** [review]

**Pull request merged**